File: app/src/main/java/igrek/songbook/about/HelpLayoutController.java
package igrek.songbook.about;

import javax.inject.Inject;

import igrek.songbook.R;
import igrek.songbook.dagger.DaggerIoc;
import igrek.songbook.info.UiInfoService;
import igrek.songbook.info.UiResourceService;

public class HelpLayoutController {
	
	@Inject
	UiInfoService uiInfoService;
	@Inject
	UiResourceService uiResourceService;
	
	public HelpLayoutController() {
		DaggerIoc.getFactoryComponent().inject(this);
	}
	
	public void showUIHelp() {
		String message = uiResourceService.resString(R.string.ui_help_content);
		String title = uiResourceService.resString(R.string.nav_help);
		uiInfoService.showDialog(title, message);
	}
}

File: app/src/main/java/igrek/songbook/about/HelpLayoutController.kt
package igrek.songbook.about

import igrek.songbook.R
import igrek.songbook.info.UiInfoService
import igrek.songbook.inject.LazyExtractor
import igrek.songbook.inject.LazyInject
import igrek.songbook.inject.appFactory

class HelpLayoutController(
        uiInfoService: LazyInject<UiInfoService> = appFactory.uiInfoService,
) {
    private val uiInfoService by LazyExtractor(uiInfoService)

    fun showUIHelp() {
        uiInfoService.dialog(R.string.nav_help, R.string.ui_help_content)
    }
}
